File: features/advisor/src/main/java/com/nivasafinance/features/advisorlead/repository/AdvisorLeadMappingRepository.java
package com.nivasafinance.features.advisorlead.repository;

import com.nivasafinance.features.advisorlead.entity.AdvisorLeadMapping;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.UUID;

@Repository
public interface AdvisorLeadMappingRepository extends JpaRepository<AdvisorLeadMapping, UUID> {
    Page<AdvisorLeadMapping> findAllByAdvisorId(UUID advisorId, Pageable pageable);
}

File: features/advisor/src/main/java/com/nivasafinance/features/advisorlead/repository/AdvisorLeadMappingRepository.kt
package com.nivasafinance.features.advisorlead.repository

import com.nivasafinance.features.advisorlead.entity.AdvisorLeadMapping
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface AdvisorLeadMappingRepository : JpaRepository<AdvisorLeadMapping, UUID> {
    fun findAllByAdvisorId(advisorId: UUID, pageable: Pageable): Page<AdvisorLeadMapping>
}
